src/isar/state_machine/transitions/robot_status.py

```diff
@@ -0,0 +1,73 @@
+from typing import TYPE_CHECKING, List
+
+from isar.state_machine.transitions.functions.robot_status import (
+    is_available,
+    is_blocked_protective_stop,
+    is_home,
+    is_offline,
+)
+from isar.state_machine.transitions.functions.utils import def_transition
+
+if TYPE_CHECKING:
+    from isar.state_machine.state_machine import StateMachine
+
+
+def get_robot_status_transitions(state_machine: "StateMachine") -> List[dict]:
+    robot_status_transitions: List[dict] = [
```

Comment:
This does kind of seem like an abuse of the conditional transitions functionality, but if it works then I guess it's fine.

Reply:
It does however seem to take away from the overall functionality of having a state machine, since here it is not state machine describing the state of the system, but the robot interface implementation doing so. It seems like the logic is a bit reversed in my opinion to make this the default method of updating the state, and I would argue it should be the state machine impacting the robot status and not the other way around. Again, I will not let this stop the PR though.

Reply:
Agreed, let's address in later refactor